db/migrate/20200310101357_add_statusand_exceeded_education_period_to_student.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class AddStatusandExceededEducationPeriodToStudent < ActiveRecord::Migration[6.0]
+  def change
+    add_column :students, :status, :integer, default: 1
+    add_column :students, :exceeded_education_period, :boolean, default: false
+    
+    add_null_constraint :students, :status 
+    add_numericality_constraint :students, :status, greater_than_or_equal_to: 0
+
+    remove_null_constraint :students, :permanently_registered
```

Comment:
`permanently_registered`'ı silmemiz lazım. Sistemde kalıcı kayıtlanmama durumu var (Öğrencinin bilgi eksikliği veya askerlik durumu ile  ilgili). Öğrenci prospective'likten öğrenciliği geçerken bu durumu `passive` durum olarak belirlemişsin bu permanently_registered durumunu yönetemiyor. Bu durum için özel bir durum belirleyebiliriz veya `permanently_registered` korunabilir.

Reply:
`permanently_registered`, öğrencinin aktif olup olmamasıyla da ilişkilendirilmişti bundan dolayı bunu kaldırıp kişinin kaydı tam yapılmadıysa pasif olarak kaydederiz diye düşünmüştüm. Açıkçası başka hangi durumlarda öğrenci pasif yapılıyor bilmiyorum. Birden fazla durum için pasif yapılıyorsa dediğin gibi kalıcı kayıtlanmama durumunu karşılamaz. O halde ben bu alanı bırakıp, kesin kayıtlı olmayanların durumunu da pasif yapıyorum.